generate_markdown_from_doxygen_xml.py

```diff
@@ -964,47 +962,51 @@ def markdown(self):
 
 
 
-
-for name in FILES_TO_PROCESS:
-    current_filename=DOXYGEN_XML_DIR+'/'+name
-
-    if name[-4:]!='.xml':
-       print(" %s - skipping (not XML)" % current_filename)
-       continue 
-    if name[-7:]=='_8h.xml':
-       print(" %s - skipping (header file)" % current_filename)
-       continue 
-    if name[-9:]=='_8cpp.xml':
-       print(" %s - skipping (cpp source file)" % current_filename)
-       continue
-    if name[:5]=='class':
-        print("  Generating: %s (class xml)" % current_filename)
-        currentClass=cppClass()
-        currentClass.import_doxygen_class_file(name,current_filename)
-        #exit()
-        markdown_string=currentClass.markdown()
-        #print(markdown_string)
-        outputfile_name=DOXYGEN_OUTPUT_DIR+'/'+name[:-4]+'.md'
+for root, dirs, files in os.walk(DOXYGEN_XML_DIR, topdown=False):
+    for name in files:
+        current_filename = os.path.join(root, name)
+
+        if name[-4:]!='.xml':
```

Comment:
For another time, you could also use `if name.endswith('.xml'):`.

Reply:
Yes, much better. I'll change that too.